schema.go

```diff
@@ -48,14 +49,37 @@ func (s *Schema[T]) Refine(predicate func(T) bool) *Schema[T] {
 	return s
 }
 
+func (s *Schema[T]) _IsOptional() bool {
+	if s.isOptional == nil {
+		return false
+	}
+	return *s.isOptional
+}
+
+func (s *Schema[T]) Optional() *Schema[T] {
+	a := true
+	s.isOptional = &a
+	return s
+}
+
 func (s *Schema[T]) Parse(value any) *ValidationResult {
```

Comment:
Based on my comments within the test file. We either need to support pointers first, then come back in and add the Optional function, or support both within the same PR. Some work needs to happen within this function to support the use-case mentioned in the test.

Reply:
Yeah I agree, these could be done as part of the same PR but realistically are two separate features.
I'll change the logic so I support pointer values for types optional or not

Reply:
That all being said, one thing to consider could be the inclusion of some Pointer() function on the schema, which would technically be outside of the scope of a zod-like schema, so may not be relevant to this project. However, this could be more useful for a go validator.